kong/plugins/moesif/regex_config_helpers.lua

```diff
@@ -0,0 +1,180 @@
+-- @return `sample_rate, block`      Sample rate and boolean flag (block or not)
+function _M.fetch_sample_rate_block_request_on_regex_match(gr_regex_configs, request_config_mapping)
+    -- Iterate through the list of governance rule regex configs
+    for _, regex_rule in pairs(gr_regex_configs) do
+        -- Fetch the sample rate
+        local sample_rate = regex_rule["sample_rate"]
+        -- Fetch the conditions
+        local conditions = regex_rule["conditions"]
+        -- Bool flag to determine if the regex conditions are matched
+        local regex_matched = nil 
+        -- Create a table to hold the conditions mapping (path and value)
+        local condition_table = {}
+
+        -- Iterate through the regex rule conditions and map the path and value
+        for _, condition in pairs(conditions) do
+            -- Check if the path does not exists already in the table (when multiple path like request.route exist in the conditions) 
+            if condition_table[condition["path"]] ~= nil then 
+                -- Create a table to hold the existing values
+                local append_value = {}
+                -- Check if the path if of type table
+                if type(condition_table[condition["path"]]) == "table" then
+                    -- Iterate through the path values and append value 
+                    for _, value in ipairs(condition_table[condition["path"]]) do 
+                        table.insert(append_value, value)
+                    end
+                else 
+                    -- Append existing value to the table
+                    table.insert(append_value, condition_table[condition["path"]])
+                end
+                -- Append new value to the existing values
+                table.insert(append_value, condition["value"])
+                -- Add condition path -> values (new and existing value) to the condition table
+                condition_table[condition["path"]] = append_value
+            else
+                -- Add condition path -> value to the condition table
+                condition_table[condition["path"]] = condition["value"]
+            end
+        end
+
+        -- Iterate through conditions table and perform `and` operation between each conditions
+        for path, values in pairs(condition_table) do 
+            -- Check if the path exists in the request config mapping
+            if request_config_mapping[path] ~= nil then 
+                -- Fetch the value of the path in request config mapping
+                local event_data = request_config_mapping[path]
+                -- Check if the condition values is of type table and not nil
+                if type(values) == 'table' and next(values) ~= nil then 
+                    -- Iterate through all the values and perform regex matching with event value
+                    for _, value in pairs(values) do 
+                        regex_matched = regex_match(event_data, value, regex_matched)
```

Comment:
no..this doesn't make sense. in a regex rule you dont have to loop through all the values